datahub/dbmaintenance/management/commands/get_dnb_one_list_tier_companies.py

```diff
@@ -0,0 +1,74 @@
+from logging import getLogger
+
+from django.core.management.base import BaseCommand
+from django.db.models import Q
+
+from datahub.company.models import Company
+
+logger = getLogger(__name__)
+
+# TODO: Remove this query command when we are done with it.
+# Side note: commands of this format will only be necessary until we have a
+# readonly replica DB to perform safe queries against.
+
+
+def _get_company_ids(one_list_tier_ids, account_manager_ids):
+    """
+    Get an iterable of duns-linked company ids that match the one list tier and account manager
+    arguments. Matching companies will be determined by those that meet these filters
+    themselves, or those whose global headquarters do.
+    """
+    tier_matches = Q(one_list_tier_id__in=one_list_tier_ids)
+    headquarter_tier_matches = Q(global_headquarters__one_list_tier_id__in=one_list_tier_ids)
+
+    companies = Company.objects.filter(
+        duns_number__isnull=False,
+        archived=False,
+    ).filter(tier_matches | headquarter_tier_matches)
+
+    if account_manager_ids:
+        account_owner_matches = Q(one_list_account_owner_id__in=account_manager_ids)
+        headquarter_account_owner_matches = Q(
+            global_headquarters__one_list_account_owner_id__in=account_manager_ids,
+        )
+        companies = companies.filter(
+            account_owner_matches | headquarter_account_owner_matches,
+        )
+
+    ids = [str(result[0]) for result in companies.values_list('id')]
```

Comment:
It looks like just:
```suggestion
    return companies.values_list('id', flat=True)
```

will work as the caller calls `str()` on each value. But will leave it up to you as this is just a reversion.